### ISSUE TITLE
[wpe-2.28] CSS - `drop-shadow` filter problems

### ISSUE THREAD

**Issue by Scony** [Metrological/WebKitForWayland#937]
In BBC test suite there are 2 test cases using `drop-shadow` filter in different contexts:
https://certification.bbctvapps.co.uk/act/test/ui/187/css-supports-shadows/
https://certification.bbctvapps.co.uk/act/test/ui/187/css-supports-filter/
In both cases, latest WPEWebKit 2.28 (and even latest https://github.com/WebKit/WebKit/commit/3142453d3762d579cda8a4f04041598e31e584ab) randomly yield wrong results.

This is expected outcome:

![Screenshot from 2022-09-12 13-01-28](https://user-images.githubusercontent.com/1207385/189642822-83c95829-e873-4b1a-b355-28ba5d600479.png)
![Screenshot from 2022-09-12 12-58-23](https://user-images.githubusercontent.com/1207385/189642880-e15f4275-ebc0-4967-9685-a40fbcd9ecf0.png)

And that's how it looks like occasionally:
![Screenshot from 2022-09-12 13-07-49](https://user-images.githubusercontent.com/1207385/189642993-41753616-f75a-409a-b309-6d4416186755.png)
![Screenshot from 2022-09-12 13-07-36](https://user-images.githubusercontent.com/1207385/189643058-28cec9f1-737c-4ad9-8257-cd983f276e03.png)

To me it looks like some compositing issues - once the CSS properties are poked a bit from web inspector, the shadow is being painted correctly.

**Comment by magomez** [comment]
For the record, this is being handled upstream in https://bugs.webkit.org/show_bug.cgi?id=245773

**Comment by magomez** [comment]
So I was able to create a simple testcase and reported this upstream as the problem is reproducible there as well (also in Safari). Apple folks told me that it was a duplicated of another issue. I tested the WiP patch that they have in that original issue and it fixed my testcase. I was going to backport that fix to 2.28 when I found that, while the patch fixes my testcase, it doesn't fix the tests reported here :(

So I'm now investigating what's the difference between my testcase and the tests here.

**Comment by pgorszkowski-igalia** [comment]
Just some notes which I gathered:
1. It still does not work in upstream
2. It works a little bit better with https://github.com/WebKit/WebKit/pull/12797/commits/9ee234dfdeeb2ab91b7c0e8124d7f981e2039918 but still it is not rendered correctly without refreshing
3. Simple test case for this issue(limit to displaying only image with filter dropShadow)
```
<html>
<head>
<style>
img {
  width: 200px;
  filter: drop-shadow(16px 16px 10px black);
}
</style>
</head>
<body>
<img src="https://certification.bbctvapps.co.uk/act/test/ui/187/css-supports-shadows/iplayerImage.png">
</body>
</html>
```

**Comment by filipe-norte-red** [comment]
Hello @magomez 
Is there any update on a possible fix for this issue? It is happening with wpe-2.38 as well. I checked related upstream issues and they are still open.
Thanks

**Comment by magomez** [comment]
Yes, this is still and issue in 2.38 and 2.42, as the upstream bug is still open. Seems that there's a PR but has some issues. I'll try to gather info from the upstream devs about it.

**Comment by filipe-norte-red** [comment]
@magomez , did you the chance to get some further info on this issue? Thanks

**Comment by magomez** [comment]
> @magomez , did you the chance to get some further info on this issue? Thanks

No progress on the upstream issue yet :(
